.github/workflows/pandora-ci.yml

```diff
@@ -25,28 +25,25 @@ concurrency:
 
 jobs:
   unit-tests:
-    permissions: write-all
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
 
     strategy:
       fail-fast: false
       max-parallel: 10
       matrix:
-        godot-version: ['4.0.4', '4.1.1', '4.2']
+        godot-version: ["4.0.4", "4.1.1", "4.2"]
 
     name: "🤖 CI on Godot ${{ matrix.godot-version }}"
-    uses: ./.github/workflows/unit-tests.yml
-    with:
-      godot-version: ${{ matrix.godot-version }}
-
-  finalize:
-    if: ${{ !cancelled() }}
-    runs-on: ubuntu-latest
-    name: Final Results
-    needs: [unit-tests]
     steps:
-      - run: exit 1
-        if: >-
-          ${{
-               contains(needs.*.result, 'failure')
-            || contains(needs.*.result, 'cancelled')
-          }}
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Run GDUnit4 tests
+        uses: MikeSchulze/gdUnit4-action@v1.1.1
```

Comment:
Please note when using the action it will fail on forked repositories because of the nature of GitHub permission stuff.
The upload report will fail on forked repositories.
To solve this you need an extra WF to publish the reports afterward by downloading the report artifacts.
Check my WF here https://github.com/MikeSchulze/gdUnit4/blob/master/.github/workflows/ci-pr-publish-report.yml
And you need to specify to disable the automated publishing reports on the `gdunit-action`
```
          publish-report: false
          report-name: gdUnit4_Godot${{ matrix.godot-version }}-${{ matrix.godot-status }}
```